src/app/configuration/configuration.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { CouchService } from '../shared/couchdb.service';
+import { ValidatorService } from '../validators/validator.service';
+import { MatStepper } from '@angular/material';
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+  templateUrl: './configuration.component.html'
+})
+export class ConfigurationComponent implements OnInit {
+  @ViewChild('stepper') stepper: MatStepper;
+  message = '';
+  select = '';
+  loginForm: FormGroup;
+  configurationFormGroup: FormGroup;
+  contactFormGroup: FormGroup;
+  nations = [];
+
+  constructor(
+    private _formBuilder: FormBuilder,
+    private couchService: CouchService,
+    private validatorService: ValidatorService,
+    private router: Router
+  ) { }
+
+  ngOnInit() {
+    this.loginForm = this._formBuilder.group({
+      username: [ '', Validators.required ],
+      password: [ '', Validators.required ],
+      confirmPassword: [ '', Validators.required ]
+    });
+    this.configurationFormGroup = this._formBuilder.group({
+      planet_type: [ '', Validators.required ],
+      local_domain: [ '', Validators.required ],
```

Comment:
You should initialize this to `document.baseURI` since that's what it ends up as in either case of Nation or Community